flakey parsing in java

I got this failure which disappeared on a re-run.

```
[info] - can extract a was failing file *** FAILED ***
[info]   com.github.javaparser.ParseProblemException: (line 8,col 11) Parse error. Found "{", expected one of  "!" "(" "\"\"\"" "class" "false" "interface" "new" "null" "super" "this" "true" "~" <CHARACTER_LITERAL> <FLOATING_POINT_LITERAL> <INTEGER_LITERAL> <LONG_LITERAL> <STRING_LITERAL>
[info] Problem stacktrace : 
[info]   com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13739)
[info]   com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13584)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceDeclaration(GeneratedJavaParser.java:468)
[info]   com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:152)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:125)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:305)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.extractString(JavaSourceEntityExtractorTest.scala:14)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.$anonfun$new$3(JavaSourceEntityExtractorTest.scala:52)
[info]   munit.internal.console.StackTraces$.dropOutside(StackTraces.scala:12)
[info]   munit.ValueTransforms.$anonfun$munitValueTransform$3(ValueTransforms.scala:37)
[info]   scala.util.Try$.apply(Try.scala:210)
[info]   munit.ValueTransforms.munitValueTransform(ValueTransforms.scala:37)
[info]   munit.ValueTransforms.munitValueTransform$(ValueTransforms.scala:22)
[info]   munit.FunSuite.munitValueTransform(FunSuite.scala:11)
[info]   munit.FunSuite.$anonfun$test$1(FunSuite.scala:37)
[info]   munit.MUnitRunner.$anonfun$runTestBody$1(MUnitRunner.scala:296)
[info]   munit.internal.console.StackTraces$.dropOutside(StackTraces.scala:12)
[info]   munit.MUnitRunner.runTestBody(MUnitRunner.scala:292)
[info]   munit.MUnitRunner.runTest(MUnitRunner.scala:264)
[info]   munit.MUnitRunner.loop$1(MUnitRunner.scala:131)
[info]   munit.MUnitRunner.$anonfun$runAsyncTestsSynchronously$1(MUnitRunner.scala:137)
[info]   munit.MUnitRunner.$anonfun$runAsyncTestsSynchronously$1$adapted(MUnitRunner.scala:137)
[info]   scala.concurrent.impl.Promise$Transformation.run(Promise.scala:470)
[info]   cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:585)
[info] (line 9,col 4) Parse error. Found <EOF>, expected one of  "!=" "%=" "&" "&&" "&=" ")" "*=" "+=" "-=" "->" "/=" "::" "<<=" "=" "==" ">>=" ">>>=" "?" "^" "^=" "|" "|=" "||"
[info] Problem stacktrace : 
[info]   com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13739)
[info]   com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13584)
[info]   com.github.javaparser.GeneratedJavaParser.PrimaryPrefix(GeneratedJavaParser.java:4762)
[info]   com.github.javaparser.GeneratedJavaParser.PrimaryExpression(GeneratedJavaParser.java:4550)
[info]   com.github.javaparser.GeneratedJavaParser.PostfixExpression(GeneratedJavaParser.java:4414)
[info]   com.github.javaparser.GeneratedJavaParser.UnaryExpressionNotPlusMinus(GeneratedJavaParser.java:4384)
[info]   com.github.javaparser.GeneratedJavaParser.UnaryExpression(GeneratedJavaParser.java:4257)
[info]   com.github.javaparser.GeneratedJavaParser.MultiplicativeExpression(GeneratedJavaParser.java:4120)
[info]   com.github.javaparser.GeneratedJavaParser.AdditiveExpression(GeneratedJavaParser.java:4072)
[info]   com.github.javaparser.GeneratedJavaParser.ShiftExpression(GeneratedJavaParser.java:4027)
[info]   com.github.javaparser.GeneratedJavaParser.RelationalExpression(GeneratedJavaParser.java:3967)
[info]   com.github.javaparser.GeneratedJavaParser.InstanceOfExpression(GeneratedJavaParser.java:3886)
[info]   com.github.javaparser.GeneratedJavaParser.EqualityExpression(GeneratedJavaParser.java:3816)
[info]   com.github.javaparser.GeneratedJavaParser.AndExpression(GeneratedJavaParser.java:3771)
[info]   com.github.javaparser.GeneratedJavaParser.ExclusiveOrExpression(GeneratedJavaParser.java:3731)
[info]   com.github.javaparser.GeneratedJavaParser.InclusiveOrExpression(GeneratedJavaParser.java:3691)
[info]   com.github.javaparser.GeneratedJavaParser.ConditionalAndExpression(GeneratedJavaParser.java:3651)
[info]   com.github.javaparser.GeneratedJavaParser.ConditionalOrExpression(GeneratedJavaParser.java:3606)
[info]   com.github.javaparser.GeneratedJavaParser.ConditionalExpression(GeneratedJavaParser.java:3574)
[info]   com.github.javaparser.GeneratedJavaParser.Expression(GeneratedJavaParser.java:3387)
[info]   com.github.javaparser.GeneratedJavaParser.ReturnStatement(GeneratedJavaParser.java:6914)
[info]   com.github.javaparser.GeneratedJavaParser.Statement(GeneratedJavaParser.java:5539)
[info]   com.github.javaparser.GeneratedJavaParser.BlockStatement(GeneratedJavaParser.java:5759)
[info]   com.github.javaparser.GeneratedJavaParser.Statements(GeneratedJavaParser.java:2667)
[info]   com.github.javaparser.GeneratedJavaParser.Block(GeneratedJavaParser.java:5638)
[info]   com.github.javaparser.GeneratedJavaParser.MethodDeclaration(GeneratedJavaParser.java:2068)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceBodyDeclaration(GeneratedJavaParser.java:1669)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceBody(GeneratedJavaParser.java:1183)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceDeclaration(GeneratedJavaParser.java:500)
[info]   com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:152)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:125)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:305)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
[info]   io.bazeltools.buildgen.javadefref.CanParseFileTest.assertParse(CanParseFileTest.scala:11)
[info]   io.bazeltools.buildgen.javadefref.CanParseFileTest.$anonfun$new$2(CanParseFileTest.scala:89)
[info]   org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[info]   org.scalatest.funsuite.AnyFunSuite.withFixture(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
[info]   org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
[info]   org.scalatest.funsuite.AnyFunSuite.runTest(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
[info]   org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   scala.collection.immutable.List.foreach(List.scala:333)
[info]   org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
[info]   org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
[info]   org.scalatest.Suite.run(Suite.scala:1114)
[info]   org.scalatest.Suite.run$(Suite.scala:1096)
[info]   org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
[info]   org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
[info]   org.scalatest.funsuite.AnyFunSuite.run(AnyFunSuite.scala:1564)
[info]   org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   sbt.TestRunner.runTest$1(TestFramework.scala:146)
[info]   sbt.TestRunner.run(TestFramework.scala:161)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.$anonfun$apply$1(TestFramework.scala:329)
[info]   sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:289)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
[info]   sbt.TestFunction.apply(TestFramework.scala:341)
[info]   sbt.Tests$.$anonfun$toTask$1(Tests.scala:435)
[info]   sbt.std.Transform$$anon$3.$anonfun$apply$2(Transform.scala:46)
[info]   sbt.std.Transform$$anon$4.work(Transform.scala:68)
[info]   sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[info]   sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[info]   sbt.Execute.work(Execute.scala:291)
[info]   sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[info]   sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[info]   sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[info]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   java.lang.Thread.run(Thread.java:750)
[info] (line 9,col 4) Parse error. Found <EOF>, expected "}"
[info] Problem stacktrace : 
[info]   com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13739)
[info]   com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13584)
[info]   com.github.javaparser.GeneratedJavaParser.Block(GeneratedJavaParser.java:5639)
[info]   com.github.javaparser.GeneratedJavaParser.MethodDeclaration(GeneratedJavaParser.java:2068)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceBodyDeclaration(GeneratedJavaParser.java:1669)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceBody(GeneratedJavaParser.java:1183)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceDeclaration(GeneratedJavaParser.java:500)
[info]   com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:152)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:125)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:305)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
[info]   io.bazeltools.buildgen.javadefref.CanParseFileTest.assertParse(CanParseFileTest.scala:11)
[info]   io.bazeltools.buildgen.javadefref.CanParseFileTest.$anonfun$new$2(CanParseFileTest.scala:89)
[info]   org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[info]   org.scalatest.funsuite.AnyFunSuite.withFixture(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
[info]   org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
[info]   org.scalatest.funsuite.AnyFunSuite.runTest(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
[info]   org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   scala.collection.immutable.List.foreach(List.scala:333)
[info]   org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
[info]   org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
[info]   org.scalatest.Suite.run(Suite.scala:1114)
[info]   org.scalatest.Suite.run$(Suite.scala:1096)
[info]   org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
[info]   org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
[info]   org.scalatest.funsuite.AnyFunSuite.run(AnyFunSuite.scala:1564)
[info]   org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   sbt.TestRunner.runTest$1(TestFramework.scala:146)
[info]   sbt.TestRunner.run(TestFramework.scala:161)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.$anonfun$apply$1(TestFramework.scala:329)
[info]   sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:289)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
[info]   sbt.TestFunction.apply(TestFramework.scala:341)
[info]   sbt.Tests$.$anonfun$toTask$1(Tests.scala:435)
[info]   sbt.std.Transform$$anon$3.$anonfun$apply$2(Transform.scala:46)
[info]   sbt.std.Transform$$anon$4.work(Transform.scala:68)
[info]   sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[info]   sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[info]   sbt.Execute.work(Execute.scala:291)
[info]   sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[info]   sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[info]   sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[info]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   java.lang.Thread.run(Thread.java:750)
[info] (line 9,col 4) Parse error. Found <EOF>, expected one of  ";" "<" "@" "abstract" "boolean" "byte" "char" "class" "default" "double" "enum" "exports" "final" "float" "int" "interface" "long" "module" "native" "open" "opens" "private" "protected" "provides" "public" "record" "requires" "short" "static" "strictfp" "synchronized" "to" "transient" "transitive" "uses" "void" "volatile" "with" "yield" "{" "}" <IDENTIFIER>
[info] Problem stacktrace : 
[info]   com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13739)
[info]   com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13584)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceBody(GeneratedJavaParser.java:1197)
[info]   com.github.javaparser.GeneratedJavaParser.ClassOrInterfaceDeclaration(GeneratedJavaParser.java:500)
[info]   com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:152)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:125)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:305)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
[info]   io.bazeltools.buildgen.javadefref.CanParseFileTest.assertParse(CanParseFileTest.scala:11)
[info]   io.bazeltools.buildgen.javadefref.CanParseFileTest.$anonfun$new$2(CanParseFileTest.scala:89)
[info]   org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[info]   org.scalatest.funsuite.AnyFunSuite.withFixture(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
[info]   org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
[info]   org.scalatest.funsuite.AnyFunSuite.runTest(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
[info]   org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   scala.collection.immutable.List.foreach(List.scala:333)
[info]   org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
[info]   org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
[info]   org.scalatest.Suite.run(Suite.scala:1114)
[info]   org.scalatest.Suite.run$(Suite.scala:1096)
[info]   org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
[info]   org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
[info]   org.scalatest.funsuite.AnyFunSuite.run(AnyFunSuite.scala:1564)
[info]   org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   sbt.TestRunner.runTest$1(TestFramework.scala:146)
[info]   sbt.TestRunner.run(TestFramework.scala:161)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.$anonfun$apply$1(TestFramework.scala:329)
[info]   sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:289)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
[info]   sbt.TestFunction.apply(TestFramework.scala:341)
[info]   sbt.Tests$.$anonfun$toTask$1(Tests.scala:435)
[info]   sbt.std.Transform$$anon$3.$anonfun$apply$2(Transform.scala:46)
[info]   sbt.std.Transform$$anon$4.work(Transform.scala:68)
[info]   sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[info]   sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[info]   sbt.Execute.work(Execute.scala:291)
[info]   sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[info]   sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[info]   sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[info]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   java.lang.Thread.run(Thread.java:750)
[info]   at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:37)
[info]   at io.bazeltools.buildgen.javadefref.CanParseFileTest.assertParse(CanParseFileTest.scala:11)
[info]   at io.bazeltools.buildgen.javadefref.CanParseFileTest.$anonfun$new$2(CanParseFileTest.scala:89)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
```

there is nothing random about that test as far as I can see, so this is concerning. I wonder if there is something janky internal to the java parser that it may leave some mutable state around in some static objects that somehow live? Just a wild guess.

again: https://github.com/ianoc/bzl-gen-build/actions/runs/3715991304/jobs/6301799102

```
io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest:
==> X io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.test basic Java Example  0.398s java.lang.AssertionError: A reference was unexpectedly null.
    at com.github.javaparser.utils.Utils.assertNotNull(Utils.java:62)
    at com.github.javaparser.TokenRange.<init>(TokenRange.java:40)
    at com.github.javaparser.GeneratedJavaParserBase.range(GeneratedJavaParserBase.java:106)
    at com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:207)
    at com.github.javaparser.JavaParser.parse(JavaParser.java:125)
    at com.github.javaparser.JavaParser.parse(JavaParser.java:305)
    at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
    at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.extractString(JavaSourceEntityExtractorTest.scala:14)
    at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.struct(JavaSourceEntityExtractorTest.scala:23)
    at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.$anonfun$new$1(JavaSourceEntityExtractorTest.scala:26)
==> X io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.test an example adding an edge with a comment  0.315s com.github.javaparser.ParseProblemException: Parse error. Found  "ptsm" <IDENTIFIER>, expected one of  ";" "@" "\u001a" "abstract" "class" "default" "enum" "final" "import" "interface" "module" "native" "open" "private" "protected" "public" "record" "static" "strictfp" "synchronized" "transient" "transitive" "volatile" <EOF>
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13739)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13584)
  com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:204)
  com.github.javaparser.JavaParser.parse(JavaParser.java:125)
  com.github.javaparser.JavaParser.parse(JavaParser.java:305)
  io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
  io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.extractString(JavaSourceEntityExtractorTest.scala:14)
  io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.$anonfun$new$3(JavaSourceEntityExtractorTest.scala:52)
  munit.internal.console.StackTraces$.dropOutside(StackTraces.scala:12)
  munit.ValueTransforms.$anonfun$munitValueTransform$3(ValueTransforms.scala:37)
  scala.util.Try$.apply(Try.scala:210)
  munit.ValueTransforms.munitValueTransform(ValueTransforms.scala:37)
  munit.ValueTransforms.munitValueTransform$(ValueTransforms.scala:22)
  munit.FunSuite.munitValueTransform(FunSuite.scala:11)
  munit.FunSuite.$anonfun$test$1(FunSuite.scala:37)
  munit.MUnitRunner.$anonfun$runTestBody$1(MUnitRunner.scala:296)
  munit.internal.console.StackTraces$.dropOutside(StackTraces.scala:12)
  munit.MUnitRunner.runTestBody(MUnitRunner.scala:292)
  munit.MUnitRunner.runTest(MUnitRunner.scala:264)
  munit.MUnitRunner.loop$1(MUnitRunner.scala:131)
  munit.MUnitRunner.$anonfun$runAsyncTestsSynchronously$1(MUnitRunner.scala:137)
  munit.MUnitRunner.$anonfun$runAsyncTestsSynchronously$1$adapted(MUnitRunner.scala:137)
  scala.concurrent.impl.Promise$Transformation.run(Promise.scala:470)
  cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:585)
Unknown error
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.jj_ntk_f(GeneratedJavaParser.java:13643)
  com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:92)
  com.github.javaparser.JavaParser.parse(JavaParser.java:125)
  com.github.javaparser.JavaParser.parse(JavaParser.java:305)
  io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
  io.bazeltools.buildgen.javadefref.CanParseFileTest.assertParse(CanParseFileTest.scala:11)
  io.bazeltools.buildgen.javadefref.CanParseFileTest.$anonfun$new$1(CanParseFileTest.scala:27)
  org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
  org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
  org.scalatest.Transformer.apply(Transformer.scala:22)
  org.scalatest.Transformer.apply(Transformer.scala:20)
  org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
  org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
  org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
  org.scalatest.funsuite.AnyFunSuite.withFixture(AnyFunSuite.scala:1564)
  org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
  org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
  org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
  org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
  org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
  org.scalatest.funsuite.AnyFunSuite.runTest(AnyFunSuite.scala:1564)
  org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
  org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
  scala.collection.immutable.List.foreach(List.scala:333)
  org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
  org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
  org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
  org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
  org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
  org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
  org.scalatest.Suite.run(Suite.scala:1114)
  org.scalatest.Suite.run$(Suite.scala:1096)
  org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
  org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
  org.scalatest.SuperEngine.runImpl(Engine.scala:535)
  org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
  org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
  org.scalatest.funsuite.AnyFunSuite.run(AnyFunSuite.scala:1564)
  org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
  org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
  sbt.TestRunner.runTest$1(TestFramework.scala:146)
  sbt.TestRunner.run(TestFramework.scala:161)
  sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.$anonfun$apply$1(TestFramework.scala:329)
  sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:289)
  sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
  sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
  sbt.TestFunction.apply(TestFramework.scala:341)
  sbt.Tests$.$anonfun$toTask$1(Tests.scala:435)
  sbt.std.Transform$$anon$3.$anonfun$apply$2(Transform.scala:46)
  sbt.std.Transform$$anon$4.work(Transform.scala:68)
  sbt.Execute.$anonfun$submit$2(Execute.scala:282)
  sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
  sbt.Execute.work(Execute.scala:291)
  sbt.Execute.$anonfun$submit$1(Execute.scala:282)
  sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
  sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
  java.util.concurrent.FutureTask.run(FutureTask.java:266)
  java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  java.util.concurrent.FutureTask.run(FutureTask.java:266)
  java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
  java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
  java.lang.Thread.run(Thread.java:750)

    at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:37)
    at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.extractString(JavaSourceEntityExtractorTest.scala:14)
    at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.$anonfun$new$3(JavaSourceEntityExtractorTest.scala:52)
  + static import 0.235s
  + static asterisk import 0.024s
  + test an annotation 0.003s
[info] CanParseFileTest:
[info] - can extract a simple file *** FAILED ***
[info]   com.github.javaparser.ParseProblemException: Parse error. Found  "ptsm" <IDENTIFIER>, expected one of  ";" "@" "\u001a" "abstract" "class" "default" "enum" "final" "import" "interface" "module" "native" "open" "private" "protected" "public" "record" "static" "strictfp" "synchronized" "transient" "transitive" "volatile" <EOF>
[info] Problem stacktrace : 
[info]   com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13739)
[info]   com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13584)
[info]   com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:204)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:125)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:305)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.extractString(JavaSourceEntityExtractorTest.scala:14)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractorTest.$anonfun$new$3(JavaSourceEntityExtractorTest.scala:52)
[info]   munit.internal.console.StackTraces$.dropOutside(StackTraces.scala:12)
[info]   munit.ValueTransforms.$anonfun$munitValueTransform$3(ValueTransforms.scala:37)
[info]   scala.util.Try$.apply(Try.scala:210)
[info]   munit.ValueTransforms.munitValueTransform(ValueTransforms.scala:37)
[info]   munit.ValueTransforms.munitValueTransform$(ValueTransforms.scala:22)
[info]   munit.FunSuite.munitValueTransform(FunSuite.scala:11)
[info]   munit.FunSuite.$anonfun$test$1(FunSuite.scala:37)
[info]   munit.MUnitRunner.$anonfun$runTestBody$1(MUnitRunner.scala:296)
[info]   munit.internal.console.StackTraces$.dropOutside(StackTraces.scala:12)
[info]   munit.MUnitRunner.runTestBody(MUnitRunner.scala:292)
[info]   munit.MUnitRunner.runTest(MUnitRunner.scala:264)
[info]   munit.MUnitRunner.loop$1(MUnitRunner.scala:131)
[info]   munit.MUnitRunner.$anonfun$runAsyncTestsSynchronously$1(MUnitRunner.scala:137)
[info]   munit.MUnitRunner.$anonfun$runAsyncTestsSynchronously$1$adapted(MUnitRunner.scala:137)
[info]   scala.concurrent.impl.Promise$Transformation.run(Promise.scala:470)
[info]   cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:585)
[info] Unknown error
[info] Problem stacktrace : 
[info]   com.github.javaparser.GeneratedJavaParser.jj_ntk_f(GeneratedJavaParser.java:13643)
[info]   com.github.javaparser.GeneratedJavaParser.CompilationUnit(GeneratedJavaParser.java:92)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:125)
[info]   com.github.javaparser.JavaParser.parse(JavaParser.java:305)
[info]   io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:33)
[info]   io.bazeltools.buildgen.javadefref.CanParseFileTest.assertParse(CanParseFileTest.scala:11)
[info]   io.bazeltools.buildgen.javadefref.CanParseFileTest.$anonfun$new$1(CanParseFileTest.scala:27)
[info]   org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[info]   org.scalatest.funsuite.AnyFunSuite.withFixture(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.invokeWithFixture$1(AnyFunSuiteLike.scala:224)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTest$1(AnyFunSuiteLike.scala:236)
[info]   org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTest(AnyFunSuiteLike.scala:236)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTest$(AnyFunSuiteLike.scala:218)
[info]   org.scalatest.funsuite.AnyFunSuite.runTest(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$runTests$1(AnyFunSuiteLike.scala:269)
[info]   org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   scala.collection.immutable.List.foreach(List.scala:333)
[info]   org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTests(AnyFunSuiteLike.scala:269)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.runTests$(AnyFunSuiteLike.scala:268)
[info]   org.scalatest.funsuite.AnyFunSuite.runTests(AnyFunSuite.scala:1564)
[info]   org.scalatest.Suite.run(Suite.scala:1114)
[info]   org.scalatest.Suite.run$(Suite.scala:1096)
[info]   org.scalatest.funsuite.AnyFunSuite.org$scalatest$funsuite$AnyFunSuiteLike$$super$run(AnyFunSuite.scala:1564)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.$anonfun$run$1(AnyFunSuiteLike.scala:273)
[info]   org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.run(AnyFunSuiteLike.scala:273)
[info]   org.scalatest.funsuite.AnyFunSuiteLike.run$(AnyFunSuiteLike.scala:272)
[info]   org.scalatest.funsuite.AnyFunSuite.run(AnyFunSuite.scala:1564)
[info]   org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:321)
[info]   org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:517)
[info]   sbt.TestRunner.runTest$1(TestFramework.scala:146)
[info]   sbt.TestRunner.run(TestFramework.scala:161)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.$anonfun$apply$1(TestFramework.scala:329)
[info]   sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:289)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
[info]   sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:329)
[info]   sbt.TestFunction.apply(TestFramework.scala:341)
[info]   sbt.Tests$.$anonfun$toTask$1(Tests.scala:435)
[info]   sbt.std.Transform$$anon$3.$anonfun$apply$2(Transform.scala:46)
[info]   sbt.std.Transform$$anon$4.work(Transform.scala:68)
[info]   sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[info]   sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[info]   sbt.Execute.work(Execute.scala:291)
[info]   sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[info]   sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[info]   sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[info]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info]   java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]   java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]   java.lang.Thread.run(Thread.java:750)
[info]   at io.bazeltools.buildgen.javadefref.JavaSourceEntityExtractor$.extract(JavaSourceEntityExtractor.scala:37)
[info]   at io.bazeltools.buildgen.javadefref.CanParseFileTest.assertParse(CanParseFileTest.scala:11)
[info]   at io.bazeltools.buildgen.javadefref.CanParseFileTest.$anonfun$new$1(CanParseFileTest.scala:27)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   ...
```

--- a/language_generators/scala-defref-extractor/src/main/scala/io/bazeltools/buildgen/javadefref/JavaSourceEntityExtractor.scala
+++ b/language_generators/scala-defref-extractor/src/main/scala/io/bazeltools/buildgen/javadefref/JavaSourceEntityExtractor.scala
@@ -21,6 +21,7 @@ import com.github.javaparser.ParseProblemException
  * https://github.com/pantsbuild/pants/blob/4e7c57db992150b3fc972e684561edb8231bba3d/src/python/pants/backend/java/dependency_inference/PantsJavaParserLauncher.java#L1
  */
 object JavaSourceEntityExtractor {
+  // this is mutable, so we need to guard any access
   private[this] lazy val parser = {
     val config = new ParserConfiguration();
 
@@ -30,15 +31,16 @@ object JavaSourceEntityExtractor {
   }
 
   def extract(content: String): IO[DataBlock] = {
-    val result = parser.parse(content)
-    (if (result.isSuccessful()) {
-       IO.pure(result.getResult().get())
-     } else {
-       IO.raiseError(new ParseProblemException(result.getProblems()))
-     }).map(structureOf(_))
+    val result = parser.synchronized { parser.parse(content) }
+    if (result.isSuccessful()) {
+      val cu = result.getResult().get()
+      IO.pure(structureOf(cu))
+    } else {
+      IO.raiseError(new ParseProblemException(result.getProblems()))
+    }
   }
 
-  def structureOf(compUnit: CompilationUnit): DataBlock = {
+  private def structureOf(compUnit: CompilationUnit): DataBlock = {
     import Entity._
     // The parser is imperative and mutable, so we take a non-idiomatic
     // approach here and use mutable values to keep state

